### ISSUE TITLE
Increase minimum required Mysql version to 5.x in Piwik 3: are there benefits?

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#9107]
Would there be any advantage to increase the minimum required version to Mysql 5.1 in Piwik 3.0.0? 

After reading https://github.com/piwik/piwik/issues/9088#issuecomment-151589488 we are wondering whether Piwik 3.0 should still support Mysql 4.x. 

(in any case, Piwik LTS 2.x will support Mysql 4.1.)

**Comment by hpvd** [comment]
beside less support effort,
1) one advantage of a newer minimum version should be a better guaranteed user experience because of speed improvements from version to version (even without any code changes)
[experience from magento usage]
2) better guaranteed security is also a great advantage, because if anything happens everyone talks about security of piwik but not the old versions of related software...

both would be also good reasons to constantly rise required versions relatively close to the latest available e.g. to only 0.2 step behind the latest
(I would strongly vote for this!)

of course this is all not only relevant for mysql, but also for php (maybe one should open a new ticket for this)

If some one hast the skill and possibility to updated piwik he/she should be also be able to maintain the relating infrastructure.

**Comment by ThaDafinser** [comment]
Supporting (really) old versions in a new major version is simple a waste of time.
Like @hpvd written i also like the approach to drop old versions in a defined way.
@hpvd PHP requiredment is here https://github.com/piwik/piwik/issues/8156

Each new PHPUnit major version only support the 2 latest versions of PHP when released
> PHPUnit 5.0 is supported on PHP 5.6 and PHP 7.

I think this can also be applied to the database. Maybe not in such a strict way, but 5.1 is far away from being "fresh"
> Version 4.1: beta from June 2004, production release October 2004
> Version 5.1: production release 27 November 2008
> Version 5.5 was generally available (as of December 2010).
> MySQL 5.6 general availability was announced in February 2013
> MySQL 5.7 Development Milestone 3 was released December 2013

***Benefit***
If you avoid one single #9088 issue you already won. Nobody currently tests such old versions in development, so it's irresponsible to say it's supported.

**Comment by tsteur** [comment]
FYI: I added https://github.com/piwik/plugin-AnonymousPiwikUsageMeasurement/issues/20 but we won't get significant statistics for this I reckon since not many are using it. 

> Nobody currently tests such old versions in development, so it's irresponsible to say it's supported.

:+1: Good point. If we say it is supported, we need to actually test it

:+1:  for @hpvd arguments as well. 

I'd increase it to at least MySQL 5.1 . If someone still wants to use an older version of MySQL: That's why we now have a LTS version for Piwik 2.X .  Even WordPress requires at least MySQL 5.0+ and they still support PHP 5.2 :)

**Comment by mattab** [comment]
First step before we can increase requirement: we need to measure actually how many users are using 4.1 - covered in #9734

**Comment by tsteur** [comment]
FYI: This decision here also affects plugin developers. If we decide to still support a 12 years old MySQL 4.1 in Piwik 3 I'm not sure if we can expect from plugin developers that they will support MySQL 4.1 and that their plugin will run on all MySQL versions.

**Comment by ThaDafinser** [comment]
> in Piwik 3 I'm not sure if we can expect from plugin developers that they will support MySQL 4.1 and that their plugin will run on all MySQL versions

This is also the case for piwik 2.x.

I still dont understand how someone still thinks about supporting mysql 4.x in a new major version.

**Comment by tsteur** [comment]
FYI: There are about 52 users sending their Piwik tracking data to our demo-anonymous instance. Out of these 52, 26 users have updated the plugin already and we are recording the MySQL version.

All of them are using MySQL 5.5 or newer see https://demo-anonymous.piwik.org/index.php?module=Widgetize&action=iframe&widget=1&moduleToWidgetize=CustomVariables&actionToWidgetize=getCustomVariables&idSite=1&period=day&date=2016-03-07&disableLink=1&widget=1

**Comment by mattab** [comment]
If we increase MySQL requirements to MySQL > 5.5.3 then it would be great to also  Make Piwik database use utf8mb4 collation by default #9785 - it would improve overall security of Piwik.

**Comment by mattab** [comment]
Out of 400+ Piwik instances already sending the Mysql version number, less than 5% are not yet using MySQL 5.5+. It seems safe already that we can increase MySQL requirements to 5.5 in Piwik 3.0.0.

**Comment by sgiehl** [comment]
Piwik 3 now requires at least MySQL 5.5. I'm closing this issue. If anything left to discuss, please reopen.
